lib/Common/Memory/WaitForConcurrentCaller.h

```diff
@@ -0,0 +1,15 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#pragma once
+
+#include "Core/CommonTypedefs.h"
+
+enum WaitForConcurrentCaller : uint8
```

Comment:
Would RecyclerWaitReason be a better/more generic name in case for example WaitForConcurrentThread gets renamed?

Reply:
fixed.